src/common.rs

```diff
@@ -1,26 +1,31 @@
-#![allow(dead_code)]
+//! Common X/Y/Z-MODEM module
 
-extern crate alloc;
+#![allow(dead_code)]
 
-use alloc::boxed::Box;
-use alloc::string::String;
+use heapless::String;
 
```

Comment:
Again, need justification for using `heapless`. I was trying to reduce the amount of external deps.